src/pages/GenericHookGuides.mdx

```diff
@@ -0,0 +1,51 @@
+## Generic Guides
+
+Here you can find a collection of useful guides how to use hooks.
+
+### How to do side effect on user action
+
+Sometimes there is need to execute some side effect while triggering user action.
+For example, you need to send request when user press a button. Or more generic,
+you need to set state inside a callback. If setting state requires the original value of state,
+the code should use the variant of `useState` that passes state as argument.
+
+I this case you can do the following thing, example is taken from [StackOverflow](https://stackoverflow.com/questions/55647287/how-to-send-request-on-click-react-hooks-way)
+with some minor changes.
+
+```tsx
+export const SomeComponent = () => {
+  const [isSending, setIsSending] = useState(false);
+  const isMounted = useRef(true);
```

Comment:
IIRC, David specifically wanted to stay away from isMounted, may want to confirm with him if his POV changed on this . If we want this to be standard practice also, we may want to create a common hook for it.